GVFS/GVFS.Service/ProductUpgradeTimer.cs

```diff
@@ -20,21 +20,36 @@ public ProductUpgradeTimer(JsonTracer tracer)
 
         public void Start()
         {
-            Random random = new Random();
-            TimeSpan startTime = TimeSpan.Zero;
+            if (!GVFSEnlistment.IsUnattended(this.tracer))

```

Comment:
I think you may need to check this inside the `TimerCallback`, because the environment variable can change. I'm not sure though if an already-running process will pick that up, so it may be that someone who turns on unattended mode for the first time also needs to restart the service. Can you please test that out?

Reply:
`Environment.GetEnvironment` will only pick up those values that were available at process startup, unless they're set inside the process with `SetEnvironmentVariable`, which we don't do here. Given that, I think this is fine as-is, and consistent with how we handle the mount process deciding whether to start the background prefetch timer.

It begs the larger question of whether this configuration _should_ be environmental or managed more directly through the application, but I don't think that's in scope for this bugfix.